test.py

```diff
@@ -1,124 +1,527 @@
+        "position": 9, 
+        "path": "./components/script/dom/eventtarget.rs"
+    }, 
+    {
+        "body": "use statement is not in alphabetical order\nexpected: dom::bindings::conversions::get_dom_class\nfound: dom::bindings::codegen::Bindings::EventTargetBinding::EventTargetMethods", 
+        "position": 10, 
+        "path": "./components/script/dom/eventtarget.rs"
+    }, 
+    {
+        "body": "use statement is not in alphabetical order\nexpected: dom::browsercontext\nfound: dom::eventtarget::EventTargetTypeId", 
+        "position": 17, 
+        "path": "./components/script/dom/bindings/utils.rs"
+    }, 
+    {
+        "body": "use statement is not in alphabetical order\nexpected: dom::eventtarget::EventTargetTypeId\nfound: dom::browsercontext", 
+        "position": 18, 
+        "path": "./components/script/dom/bindings/utils.rs"
+    }
+]
+
+
+class TestAPIProvider(unittest.TestCase):
+    def setUp(self):
+        self.api_provider = APIProvider('jdm')
+
+
+    def test_is_new_contributor(self):
+        with self.assertRaises(NotImplementedError):
+            self.api_provider.is_new_contributor("jdm")
+
+
+    def test_post_comment(self):
+        with self.assertRaises(NotImplementedError):
+            self.api_provider.post_comment("Nice job!", 3947)
+
+
+    def test_post_review_comment(self):
+        with self.assertRaises(NotImplementedError):
+            self.api_provider.post_review_comment(1234, "a453b3923e893f0383cd2893f...", "foo/bar/spam/eggs", 3, "Remove extra space")
+
+
+    def test_add_label(self):
+        with self.assertRaises(NotImplementedError):
+            self.api_provider.add_label("S-awaiting-review", 1234)
+
+
+    def test_remove_label(self):
+        with self.assertRaises(NotImplementedError):
+            self.api_provider.remove_label("S-awaiting-review", 1234)
+
+
+    def test_get_labels(self):
+        with self.assertRaises(NotImplementedError):
+            self.api_provider.get_labels(1234)
+
+
+    def test_get_diff(self):
+        with self.assertRaises(NotImplementedError):
+            self.api_provider.get_diff("https://github.com/servo/servo/pull/1234.diff")
+
+
+    def test_set_assignee(self):
+        with self.assertRaises(NotImplementedError):
+            self.api_provider.set_assignee("jdm", 1234)
+
+
+# TODO - add tests for exception handling
+class TestGithubApiProvider(unittest.TestCase):
+    class FakeHeader():
+            def get(self, something):
+                pass
+
+    def setUp(self):
+        self.owner = "servo"
+        self.repo = "servo"
+        self.gh_provider = GithubApiProvider("jdm", "a453b3923e893f0383cd2893f...", self.owner, self.repo)
+        self.gh_provider.api_req = Mock()
+
+    @patch('githubapiprovider.gzip')
+    @patch('githubapiprovider.urllib2')
+    @patch('githubapiprovider.base64')
+    def test_api_req(self, base64_mock, urllib_mock, gzip_mock):
```

Comment:
This feels like it's testing things that aren't important to verify. When we're mocking the internals of opening a URL, I think we've gone too far down the testing rabbit hole.